docs/src/man/basics.md

```diff
@@ -0,0 +1,1220 @@
+```
+
+## Using `select`, and `select!`, `transform`, and `transform!`, `sort!`
+
+You can also use the `select` and `select!` functions to select, rename, and transform columns in a DataFrame.
+
+The `select` function creates a new DataFrame:
+
+```jldoctest dataframe
+julia> german = CSV.read((joinpath(dirname(pathof(DataFrames)),
+                                 "..", "docs", "src", "assets", "german.csv")),
+                       DataFrame);
+
+julia> select(german, Not(:Age)) # drop column :x1 in a new data frame
+1000×9 DataFrame
+  Row │ id     Sex     Job    Housing  Saving accounts  Checking account  Cred ⋯
+      │ Int64  String  Int64  String   String           String            Int6 ⋯
+──────┼─────────────────────────────────────────────────────────────────────────
+    1 │     0  male        2  own      NA               little                 ⋯
+    2 │     1  female      2  own      little           moderate
+    3 │     2  male        1  own      little           NA
+    4 │     3  male        2  free     little           little
+    5 │     4  male        2  free     little           little                 ⋯
+    6 │     5  male        1  free     NA               NA
+    7 │     6  male        2  own      quite rich       NA
+    8 │     7  male        3  rent     little           moderate
+  ⋮   │   ⋮      ⋮       ⋮       ⋮            ⋮                ⋮               ⋱
+  994 │   993  male        3  own      little           little                 ⋯
+  995 │   994  male        2  own      NA               NA
+  996 │   995  female      1  own      little           NA
+  997 │   996  male        3  own      little           little
+  998 │   997  male        2  own      little           NA                     ⋯
+  999 │   998  male        2  free     little           little
+ 1000 │   999  male        2  own      moderate         moderate
+                                                  3 columns and 985 rows omitted
+
+julia> select(german, r"Sex") # select columns containing 'x' character
+1000×1 DataFrame
+  Row │ Sex
+      │ String
+──────┼────────
+    1 │ male
+    2 │ female
+    3 │ male
+    4 │ male
+    5 │ male
+    6 │ male
+    7 │ male
+    8 │ male
+  ⋮   │   ⋮
+  994 │ male
+  995 │ male
+  996 │ female
+  997 │ male
+  998 │ male
+  999 │ male
+ 1000 │ male
+985 rows omitted
+
+julia> select(german, :Sex => :x1, :Age => :x2) # rename columns
+1000×2 DataFrame
+  Row │ x1      x2
+      │ String  Int64
+──────┼───────────────
+    1 │ male       67
+    2 │ female     22
+    3 │ male       49
+    4 │ male       45
+    5 │ male       53
+    6 │ male       35
+    7 │ male       53
+    8 │ male       35
+  ⋮   │   ⋮       ⋮
+  994 │ male       30
+  995 │ male       50
+  996 │ female     31
+  997 │ male       40
+  998 │ male       38
+  999 │ male       23
+ 1000 │ male       27
+      985 rows omitted
+
+julia> select(german, :Age, :Age => ByRow(sqrt)) # transform columns by row
```

Comment:
maybe give some simpler examples first.

Reply:
Please suggest, I was not getting a simpler example.

Reply:
a transformation that is not by-row but takes whole column